src/v/model/fundamental.h

```diff
@@ -89,6 +91,22 @@ class topic : public named_type<ss::sstring, struct model_topic_type> {
     topic(model::topic_view view) // NOLINT - see topic_view_tests.cc
       : named_type<ss::sstring, struct model_topic_type>(ss::sstring(view())) {}
 
+    friend void
+    read_nested(iobuf_parser& in, topic& el, size_t const bytes_left_limit) {
+        using serde::read_nested;
+        return read_nested(
+          in,
+          *static_cast<named_type<ss::sstring, struct model_topic_type>*>(&el),
```

Comment:
instead of the cast

Reply:
I changed it to `el._value` since `named_type` is only for the compiler to check types but for binary serialization this will probably never matter. So just serializing the underlying type directly should be fine.

Reply:
great, thank you